### PR TITLE
use latest controllers and webhook in the deployment

### PR DESCRIPTION
**Pull request by humblec**
- [x] example restore yaml corrected to point to the snapshot object created from this repo 
- [x] Fix a couple of CVEs on dependent packages

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

> /kind cleanup

-->
```release-note
NONE
```

### CODE REVIEW

File: deploy/kubernetes/csi-snapshotter/setup-csi-snapshotter.yaml [hunk 1]
```diff
@@ -72,7 +72,7 @@ spec:
       serviceAccountName: csi-snapshotter
       containers:
         - name: csi-provisioner
-          image: k8s.gcr.io/sig-storage/csi-provisioner:v3.0.0
+          image: k8s.gcr.io/sig-storage/csi-provisioner:v3.1.0
```
Comment (xing-yang) [inline deploy/kubernetes/csi-snapshotter/setup-csi-snapshotter.yaml:75]: Let's wait for the upcoming csi-provisioner release which depends on snapshotter 6.0.1

Comment (humblec) [reply deploy/kubernetes/csi-snapshotter/setup-csi-snapshotter.yaml:75]: sure , we can wait

Comment (humblec) [reply deploy/kubernetes/csi-snapshotter/setup-csi-snapshotter.yaml]: @xing-yang updated to v3.2.0 ... :+1:

File: deploy/kubernetes/csi-snapshotter/setup-csi-snapshotter.yaml [hunk 2]
```diff
@@ -97,7 +97,7 @@ spec:
             - name: socket-dir
               mountPath: /csi
         - name: hostpath
-          image: k8s.gcr.io/sig-storage/hostpathplugin:v1.7.2
+          image: k8s.gcr.io/sig-storage/hostpathplugin:v1.8.0
```
Comment (xing-yang) [inline deploy/kubernetes/csi-snapshotter/setup-csi-snapshotter.yaml:100]: Let's wait for a new hostpathplugin image.

Comment (humblec) [reply deploy/kubernetes/csi-snapshotter/setup-csi-snapshotter.yaml:100]: sure.

**Comment by xing-yang** [comment]
Sorry, I forgot about this PR and merged another similar one: https://github.com/kubernetes-csi/external-snapshotter/pull/737.

Please use this PR to update CSI hostpath driver version after that becomes available.

**Comment by humblec** [comment]
> Sorry, I forgot about this PR and merged another similar one: #737.
> 
> Please use this PR to update CSI hostpath driver version after that becomes available.

nw, I rebased to master and the only change is in the example yaml now.. we can get this in as it is as sidecar update is no longer part of this pr @xing-yang

**Comment by xing-yang** [comment]
There's a Trivy failure: https://github.com/kubernetes-csi/external-snapshotter/runs/7297605065?check_suite_focus=true
Can you try to update the lib github.com/emicklei/go-restful from v2.9.5 to v3.8.0 as suggested?
https://github.com/kubernetes-csi/external-snapshotter/blob/master/go.mod#L36

**Comment by humblec** [comment]
> @humblec There's a Trivy failure: https://github.com/kubernetes-csi/external-snapshotter/runs/7297605065?check_suite_focus=true Can you try to update the lib github.com/emicklei/go-restful from v2.9.5 to v3.8.0 as suggested? https://github.com/kubernetes-csi/external-snapshotter/blob/master/go.mod#L36

@xing-yang its an indirect dependency update and iic, its coming from client-go/discovery and  kube-openapi.. client-go looks to be using the v3 of this package in master https://github.com/kubernetes/client-go/blob/master/go.mod#L39 also kube-openapi https://github.com/kubernetes/kube-openapi/blob/master/go.mod#L9 . In this PR's last commit I have updated the client go and kube-openapi to latest which take care of the same.. :+1:

**Comment by xing-yang** [comment]
Thanks @humblec for taking care of the Trivy failures!

/lgtm
/approve

**Pull request merged**